packages/react/src/file-upload/FileUpload.tsx

```diff
@@ -106,7 +110,32 @@ export const FileUpload = forwardRef<HTMLInputElement, FileUploadProps>(
 
 		const handleRemoveFile = (file: FileWithStatus) => {
 			const indexOfFile = value.indexOf(file);
-			onChange(value.filter((_, index) => index !== indexOfFile));
+			const newBaseSetOfFiles = value.filter(
+				(_, index) => index !== indexOfFile
+			);
+			const hasTooManyFilesError = fileRejections.some((rej) =>
+				rej.errors.some((err) => err.code === TOO_MANY_FILES_ERROR.code)
+			);
+			if (hasTooManyFilesError) {
+				const acceptableFiles = fileRejections.filter((rej) =>
+					rej.errors.every((err) => err.code === TOO_MANY_FILES_ERROR.code)
+				);
+
+				if (acceptableFiles.length > 0) {
+					const acceptedFile = acceptableFiles[0].file;
+					const newFiles = [...newBaseSetOfFiles, acceptedFile].sort(
```

Comment:
I don't think we should sort. If I upload 1.txt, 3.txt, 5.txt, 2.txt, 4.txt, with 2 and 4 being over the limit, then remove 1, I do not expect the order to be 2, 3, 5, 4. It should maintain the order that they were dropped in the original list

Reply:
Sorting functionality has been removed. I also removed the sorting util functions and tests since they were not being used anymore.